src/puppetlabs/puppetdb/client.clj

```diff
@@ -6,78 +6,80 @@
             [clj-http.client :as http-client]
             [puppetlabs.puppetdb.command.constants :refer [command-names]]
             [puppetlabs.puppetdb.cheshire :as json]
-            [puppetlabs.kitchensink.core :as kitchensink]))
+            [puppetlabs.puppetdb.schema :refer [defn-validated]]
+            [puppetlabs.puppetdb.utils :as utils]
+            [puppetlabs.kitchensink.core :as kitchensink]
+            [schema.core :as s]))
 
-(defn submit-command-via-http!
+(defn-validated submit-command-via-http!
   "Submits `payload` as a valid command of type `command` and
   `version` to the PuppetDB instance specified by `host` and
   `port`. The `payload` will be converted to JSON before
   submission. Alternately accepts a command-map object (such as those
   returned by `parse-command`). Returns the server response."
-  ([host port command version payload]
-     {:pre [(string? command)
-            (integer? version)]}
+  ([base-url
```

Comment:
Might be useful to distinguish between "version" below, which is command version and version in base-url. Maybe just rename version to command-version?